pkg/common/catalog/plugin.go

```diff
@@ -41,6 +42,12 @@ func newPlugin(ctx context.Context, conn grpc.ClientConnInterface, info PluginIn
 		return nil, err
 	}
 
+	closers = append(closers, closerFunc(func() {
+		if err := private.Deinit(ctx, conn); err != nil {
```

Comment:
@azdagron I'm wondering if by the time that this is called, the context is already canceled?